include/openssl/kdf.h

```diff
@@ -15,80 +15,107 @@
+# define EVP_PKEY_CTRL_TLS_MD                   EVP_PKEY_CTRL_KDF_MD
+# define EVP_PKEY_CTRL_HKDF_MD                  EVP_PKEY_CTRL_KDF_MD
+# define EVP_PKEY_CTRL_HKDF_SALT                EVP_PKEY_CTRL_KDF_SALT
+# define EVP_PKEY_CTRL_HKDF_KEY                 EVP_PKEY_CTRL_KDF_KEY
+# define EVP_PKEY_CTRL_HKDF_INFO                EVP_PKEY_CTRL_KDF_INFO
+# define EVP_PKEY_CTRL_HKDF_MODE                EVP_PKEY_CTRL_KDF_MODE
 
 # define EVP_PKEY_HKDEF_MODE_EXTRACT_AND_EXPAND 0
 # define EVP_PKEY_HKDEF_MODE_EXTRACT_ONLY       1
 # define EVP_PKEY_HKDEF_MODE_EXPAND_ONLY        2
 
-# define EVP_PKEY_CTX_set_tls1_prf_md(pctx, md) \
-            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_TLS_MD, 0, (void *)(md))
-
-# define EVP_PKEY_CTX_set1_tls1_prf_secret(pctx, sec, seclen) \
-            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_TLS_SECRET, seclen, (void *)(sec))
 
-# define EVP_PKEY_CTX_add1_tls1_prf_seed(pctx, seed, seedlen) \
+/* Generic setters */
+# define EVP_PKEY_CTX_set_pbe_md(pctx, md) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_TLS_SEED, seedlen, (void *)(seed))
+                              EVP_PKEY_CTRL_KDF_MD, 0, (void *)(md))
 
-# define EVP_PKEY_CTX_set_hkdf_md(pctx, md) \
+# define EVP_PKEY_CTX_set1_pbe_pass(pctx, pass, passlen) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_HKDF_MD, 0, (void *)(md))
+                              EVP_PKEY_CTRL_KDF_PASS, passlen, (void *)(pass))
 
-# define EVP_PKEY_CTX_set1_hkdf_salt(pctx, salt, saltlen) \
+# define EVP_PKEY_CTX_set1_pbe_salt(pctx, salt, saltlen) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_HKDF_SALT, saltlen, (void *)(salt))
+                              EVP_PKEY_CTRL_KDF_SALT, saltlen, (void *)(salt))
 
-# define EVP_PKEY_CTX_set1_hkdf_key(pctx, key, keylen) \
+# define EVP_PKEY_CTX_set1_pbe_key(pctx, key, keylen) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_HKDF_KEY, keylen, (void *)(key))
+                              EVP_PKEY_CTRL_KDF_KEY, keylen, (void *)(key))
 
-# define EVP_PKEY_CTX_add1_hkdf_info(pctx, info, infolen) \
+# define EVP_PKEY_CTX_add1_pbe_info(pctx, info, infolen) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_HKDF_INFO, infolen, (void *)(info))
+                              EVP_PKEY_CTRL_KDF_INFO, infolen, (void *)(info))
 
-# define EVP_PKEY_CTX_hkdf_mode(pctx, mode) \
+# define EVP_PKEY_CTX_set_pbe_mode(pctx, mode) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_HKDF_MODE, mode, NULL)
+                              EVP_PKEY_CTRL_KDF_MODE, mode, NULL)
 
```

Comment:
There is also inconsistency between the ctrl name and the macro. I'd name these so that anything that is likely to be used by multiple EVP_PKEY_METHODs (some of which may not be KDFs) would just be of the form EVP_PKEY_CTRL_FOO with the macro EVP_PKEY_CTX_set_foo(). If it could be used by multiple KDFs (which might include PBEs) then EVP_PKEY_CTRL_KDF_FOO with macro EVP_PKEY_CTX_set_kdf_foo(), If it is a PBE only then EVP_PKEY_CTRL_PBE_FOO and EVP_PKEY_CTX_set_pbe_foo(). Finally anything algorithm specific for algorithm "alg" becomes EVP_PKEY_CTRL_ALG_FOO and EVP_PKEY_CTX_set_alg_foo().